[sdevih] iP

sdevih opened the pull request:
## DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/sdevih/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :star_struck:

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

sdevih committed:
Added Level-0

sdevih committed:
Added level-1

sdevih committed:
Added Level-2

sdevih committed:
Added Level-3 with marking as done functionality

sdevih committed:
Added Level-4 with tracking for different types of tasks

sdevih committed:
Cleaned up Level-4 with formatting

sdevih committed:
Added Automated Text UI Testing

sdevih committed:
Added Leevl-5 with error handling

sdevih committed:
Added Level-6 with delete function

sdevih committed:
Update task array to arraylist

sdevih committed:
Add save function for tasks

sdevih committed:
Merge branch-Level-7

sdevih committed:
Read date and time for tasks

sdevih committed:
Merge branch-Level-8

sdevih committed:
Refactor code to extract as classes

sdevih committed:
Organise classes into packages

sdevih committed:
Merge branch 'master' into add-gradle-support

sdevih committed:
Merge branch 'add-gradle-support'

sdevih committed:
Add JUnit tests

sdevih committed:
Package as JAR file

sdevih committed:
Add more Javadocs

sdevih committed:
Merge branch-javadocs

sdevih committed:
Edit code to follow coding standard

sdevih committed:
Add finding task by keyword function

sdevih committed:
Merge branch-Level-9

sdevih committed:
Update README.md

sdevih committed:
Add more javadocs and fix style

sdevih committed:
Merge branch-A-CodingStandard

sdevih committed:
Merge branch 'master' of https://github.com/sdevih/ip

PallonCX commented on src/main/java/seedu/chatteroo/commands/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+        int listCount = listStore.getTaskListSize();
+        System.out.println("Got it. I've added this task:\n" + newTask.toString());
+        if (listCount == 1) {
+        System.out.println("Now you have " + listCount + " task in the list.\n");
```
You may want to indent after an if statement:

```Suggestion
        if (listCount == 1) {
            System.out.println("Now you have " + listCount + " task in the list.\n");
```

PallonCX commented on src/main/java/seedu/chatteroo/tasks/Deadline.java:
```diff
@@ -0,0 +1,40 @@
+        try {
+            this.by = LocalDateTime.parse(by.trim(), dateFormat);
+        } catch (DateTimeException e) {
+            System.out.println("Please enter a valid date and time in the format yyyy-mm-dd HHmm!\n" +
+                    "For example, 2024-01-31 1800");        }
```
Will it improve the readability if you start the next line with operator?

```Suggestion
            System.out.println("Please enter a valid date and time in the format yyyy-mm-dd HHmm!\n"
                    + "For example, 2024-01-31 1800");        }
```

PallonCX commented on src/main/java/seedu/chatteroo/tasks/Event.java:
```diff
@@ -0,0 +1,48 @@
+            this.from = LocalDateTime.parse(from.trim(), dateFormat);
+            this.to = LocalDateTime.parse(to.trim(), dateFormat);
+        } catch (DateTimeException e) {
+            System.out.println("Please enter a valid date and time in the format yyyy-mm-dd HHmm!\n" +
+                    "For example, 2024-01-31 1800");
```
Will it improve the readability if you start the next line with operator?

```Suggestion
            System.out.println("Please enter a valid date and time in the format yyyy-mm-dd HHmm!\n"
                    + "For example, 2024-01-31 1800");
```

PallonCX commented on src/main/java/seedu/chatteroo/tasks/Task.java:
```diff
@@ -0,0 +1,53 @@
+    public void markAsNotdone() {
+        this.isDone = false;
+    }
+
```
You may add some JavaDoc for this method.

PallonCX reviewed (commented):
Well done, LGTM. Just some nits to fix.

officialchengyud commented on data/chatteroo.txt:
Should this be committed? You can consider adding this to .gitignore

officialchengyud commented on src/main/java/seedu/chatteroo/commands/AddCommand.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public void execute(TaskList listStore, Ui ui, Storage storage) {
```
Can consider adding some JavaDocs here for more clarity and documentation

officialchengyud commented on src/main/java/seedu/chatteroo/parser/Parser.java:
```diff
@@ -0,0 +1,90 @@
+     */
+    public static Command parseInput(String input) throws Exception {
+        String[] inputArr = input.split(" ");
+        String command = inputArr[0].toUpperCase();
```
I like how you used switch statements with correct indentation here. You may consider adding a new line after String command for more clarity

officialchengyud commented on src/main/java/seedu/chatteroo/tasks/Event.java:
```diff
@@ -0,0 +1,48 @@
+        return "[E]" + super.toString() + " (from: " + formattedFrom + " to: " + formattedTo + ")";
+    }
+
+    public LocalDateTime getFrom() {
```
You may consider JavaDoc here too

officialchengyud reviewed (commented):
Awesome job! LGTM, just some small nits regarding code quality and adding JavaDocs to certain methods.

sdevih committed:
Add GUI function

sdevih committed:
Merge branch-Level-10

Pughal77 commented on src/main/java/seedu/chatteroo/storage/Storage.java:
```diff
@@ -0,0 +1,117 @@
+                sc.close();
+            }
+        } catch (IOException e) {
+            System.out.println("ChatterOOHNOO! Chatteroo can't create a new file!");
```
Perhaps this response should be handled by the Ui?

Pughal77 commented on src/main/java/seedu/chatteroo/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```
perhaps you can inherit the java doc and elaborate more on how you are exceuting delete?

Pughal77 commented on src/main/java/seedu/chatteroo/tasks/Event.java:
```diff
@@ -0,0 +1,48 @@
+    protected String taskType = "E";
+
+    /**
+     * Constructor for the Event class.
```
There should be an empty line between the dexcription and parameters

Pughal77 commented on src/main/java/seedu/chatteroo/ui/Ui.java:
```diff
@@ -0,0 +1,57 @@
+        System.out.println("There are no tasks in the list.\n");
+    }
+
+    public String showAddTaskText(Task newTask, int listCount) {
```
Perhaps instead of naming them as showAddTaskTest you could say that it is a response to the user and name it as
showAddTaskResponse? Similarly for the rest as well.

Pughal77 reviewed (commented):
LGTM. just a few nits

sdevih committed:
Refactor code for OOP

sdevih committed:
Add assertions

Assertions were added to the Chatteroo, Parser and Task classes
to document important assumptions.

sdevih committed:
Merge pull request #1 from sdevih/branch-A-Assertions

Add assertions

sdevih committed:
Improve code quality

The long methods within Storage class were refactored into other
classes. The loadTasks method was shortened by extracting logic into a
parseTaskFiles method in Parser class. The saveTasks method was
shortened by extracting logic into a formatTasksforFile method within
Task class.

The extraction of logic helps improve the readability of the code.

Javadocs were also added to the different command classes. The
@inheritDoc tag was used to show that that the method inherits its
documentation from the command class.

sdevih committed:
Merge pull request #3 from sdevih/branch-A-CodeQuality

Improve code quality

sdevih committed:
Implement DoneCommand

The DoneCommand removes all tasks marked as done from the task list

sdevih committed:
Merge pull request #4 from sdevih/branch-C-MassOps

Implement DoneCommand

sdevih committed:
Handle more errors

sdevih committed:
Merge branch 'branch-A-MoreErrorHandling'

sdevih committed:
Improve gui

sdevih committed:
Merge branch 'branch-A-BetterGui'

sdevih committed:
Improve personality

Chatteroo responds to commands with a better personality.

The file path was also changed to remove the seedu sub-directory.

sdevih committed:
Merge branch 'branch-A-BetterPersonality'

sdevih committed:
Refactor classes

sdevih committed:
Ignore data text file

sdevih committed:
Untrack data text file

sdevih committed:
Set up product website

sdevih committed:
Add Javadocs

sdevih committed:
Update junit event test
